Add Progress Bar

amitd291 opened the pull request:
This PR achieves the following:
 - Add determinate circular progress bar to represent the time remaining in a more visual manner.

amitd291 committed:
Add determinate circular progress bar to represent the time remaining <Amit/Prasi>

amitd291 committed:
Merge branch 'master' into feature/add_progress_bar

praslnx8 commented on app/src/main/java/com/mymeetings/pairpomodoro/view/MainActivity.kt:
```diff
@@ -158,6 +170,14 @@ class MainActivity : AppCompatActivity(), ServiceConnection {
 
+    override fun onResume() {
+        super.onResume()
+        timerProgressBar.max =
```
Why this has to be executed in every onResume.

amitd291 replied:
There's a bug otherwise and the progressbar max is reset to zero. But after replacing with Android progress bar it might have been fixed. I'll test this scenario.

praslnx8 replied:
Yes. I got it. But we should get it from Timer as the TimerPreference value may vary when user go with sync mode.

praslnx8 replied:
This issue is now fixed.

praslnx8 commented on app/src/main/java/com/mymeetings/pairpomodoro/view/MainActivity.kt:
```diff
@@ -158,6 +170,14 @@ class MainActivity : AppCompatActivity(), ServiceConnection {
+        super.onResume()
+        timerProgressBar.max =
+            TimeUnit.MINUTES.toMillis(
+                PreferenceManager.getDefaultSharedPreferences(this).getInt(FOCUS_TIME_KEY, 25).toLong()
```
This should not be from sharedPreference.

amitd291 replied:
Again they progress bar max was getting reset, will check this scenario with the Android progress bar.

praslnx8 replied:
This issue is now fixed. We have to call timerCreated method again whenever our activity call service for handshake

praslnx8 commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -44,6 +44,22 @@
+            android:layout_height="wrap_content"
+            android:layout_gravity="center">
+
+        <me.zhanghai.android.materialprogressbar.MaterialProgressBar
```
Shall we use android provided ProgressBar.
Or we can use layer-list to build our own drawable.

amitd291 replied:
Saw your changes, so we should add a drawable to the progress bar to fix horizontal to circle 👍

praslnx8 commented on app/src/test/java/com/mymeetings/pairpomodoro/model/pomodoroManager/PomodoroManagerTest.kt:
```diff
@@ -36,7 +36,8 @@ class PomodoroManagerTest {
             timerAlarm = timerAlarm,
             timerSyncer = timerSyncer,
-            updateCallback = updateCallback
+            onTimerCreated = { _, _ -> },
```
Why this is empty. It should send back to view.

praslnx8 commented on app/src/main/java/com/mymeetings/pairpomodoro/view/MainActivity.kt:
```diff
@@ -75,29 +82,36 @@ class MainActivity : AppCompatActivity(), ServiceConnection {
+        timerPreference: TimerPreference,
+        sharingKey: String
+    ) {
+        timerProgressBar.max = timerPreference.getFocusTime().toInt()
```
The max should vary for different stages.
Eg. For focus -> getFocusTime
For ShortBreak -> the max should the getShortBreakTime

amitd291 replied:
Yeah correct, so we should have a type of timer to handle that?

praslnx8 replied:
Yes. Timer gives PomodoroState in PomodoroStatus. We can use that.

praslnx8 replied:
I added the logic. Please review.

praslnx8 committed:
1. Removed MaterialProgeressBar and used drawable.
2. Removed setting the max time from preference in onResume

praslnx8 committed:
Fixed pomodoro timer onCreated not called when activity doing hand shake.

praslnx8 committed:
Added logic to switch max time based on pomodoroMode.

praslnx8 committed:
Merge branch 'master' into feature/add_progress_bar

praslnx8 committed:
Fixed progress bar not updating max value issue.
Adjusted the Progresbar.

praslnx8 committed:
Fixed progress bar not updating max value issue.
Adjusted the Progresbar.

praslnx8 committed:
Fixed failing test in PomodoroManagerTest

amitd291 committed:
Make the progress bar more visible, reduce the progress units for a smoother progress visualiation on the UI

praslnx8 committed:
1. Fix sending parcellable TimerPreference instead of non parcellable object.
2. Set notification sound as null to show silent notification.
3. make the progress thinner for UI.
